contracts/strategies/idle/StrategyIdleLendingPool.sol

```diff
@@ -0,0 +1,246 @@
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./GovTokenRegistry.sol";
+
+import "../interfaces/IStrategy.sol";
+import "../interfaces/idle/IIdleToken.sol";
+import "../interfaces/aave/IStakedAave.sol";
+import "../interfaces/uniswap/IUniswapV2.sol";
+
+/**
+ * Deposits ERC20 token into Idle Lending Pool V4. Holds IdleErc20(e.g. IdleDAI, IdleUSDC).
+ */ 
+contract StrategyIdleLendingPool is IStrategy, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // Governance token registry
+    GovTokenRegistry govTokenRegistry;
+
+    // The address of Idle Lending Pool(e.g. IdleDAI, IdleUSDC)
+    address public iToken;
+   
+    // Info of supplying erc20 token to Aave lending pool
+    // The symbol of the supplying token
+    string public symbol;
+    // The address of supplying token (e.g. DAI, USDT)
+    address public supplyToken;
+    // Supplying token decimals
+    uint8 public decimals;
+    
+    // The address of Aave StakedAave contract
+    address public stakedAave;
+
+    address public weth;
+    address public sushiswap;
+
+    address public controller;
+
+    uint256 constant public FULL_ALLOC = 100000;
+
+    constructor(
+        address _iToken,
+        string memory _symbol,
+        address _supplyToken,
+        uint8 _decimals,
+        address _govTokenRegistryAddress,
+        address _stakedAave,
+        address _weth,
+        address _sushiswap,
+        address _controller
+    ) {
+        iToken = _iToken;
+        symbol = _symbol;
+        supplyToken = _supplyToken;
+        decimals = _decimals;
+        govTokenRegistry = GovTokenRegistry(_govTokenRegistryAddress);
+        stakedAave = _stakedAave;
+        weth = _weth;
+        sushiswap = _sushiswap;
+        controller = _controller;
+    }
+
+    /**
+     * @dev Require that the caller must be an EOA account to avoid flash loans.
+     */
+    modifier onlyEOA() {
+        require(msg.sender == tx.origin, "Not EOA");
+        _;
+    }
+
+    function getAssetAddress() external view override returns (address) {
+        return supplyToken;
+    }
+
+    function harvest() external override onlyEOA {
+        // Claim governance tokens without redeeming supply token
+        IIdleToken(iToken).redeemIdleToken(uint256(0));
+
+        harvestAAVE();
+        swapGovTokensToSupplyToken();
+
+        // Deposit obtained supply token to Idle Lending Pool
+        uint256 obtainedSupplyTokenAmount = IERC20(supplyToken).balanceOf(address(this));
+        IERC20(supplyToken).safeIncreaseAllowance(iToken, obtainedSupplyTokenAmount);
+        IIdleToken(iToken).mintIdleToken(obtainedSupplyTokenAmount, false, address(0));
+    }
+
+    function syncBalance() external view override returns (uint256) {
+        uint256 iTokenPrice = IIdleToken(iToken).tokenPrice();
```

Comment:
Hey, I'm one of the founders and lead dev of Idle finance. I stumbled upon this integration, and I would like to suggest in general to always use `tokenPriceWithFee` instead of the plain `tokenPrice`. See here for more info https://developers.idle.finance/methods/tokenpricewithfee . In this way you can calculate your 'real' balance (ie with fee already counted, so you will have a net token price)

Reply:
In this way you can also substitute the `getRedeemPrice` potentially

Reply:
Hi, bugduino. L2 finance will also integrate IDLE risk adjusted token which does not have `tokenPriceWithFee` method so I replace the `getReedmPrice` with above ref method. Thank you for your advice!

Reply:
Oh ok ok, sorry was thinking about the Best yield! Nice catch then :+1: